packages/rust-dprint-plugin-typescript/src/main.rs

```diff
@@ -0,0 +1,189 @@
+        ))
+    }
+}
+
+fn format_source_files(config: Configuration, paths: Vec<PathBuf>) -> Result<(), std::io::Error> {
+    let mut not_formatted_files = vec![];
+
+    for file_path in paths {
+        let file_path_str = file_path.to_string_lossy();
+        let file_contents = fs::read_to_string(&file_path)?;
+
+        match dprint::format_text(&file_path_str, &file_contents, &config) {
+            Ok(None) => {
+                // nothing to format, pass
+            }
+            Ok(Some(formatted_text)) => {
+                if formatted_text != file_contents {
+                    println!("{}", file_path_str);
+                    fs::write(&file_path, formatted_text)?;
+                    not_formatted_files.push(file_path);
+                }
+            }
+            Err(e) => {
+                eprintln!("Error formatting: {}", &file_path_str);
+                eprintln!("   {}", e);
+            }
+        }
+    }
+
+    let f = if not_formatted_files.len() == 1 {
+        "file"
+    } else {
+        "files"
+    };
+    println!("Formatted {} {}", not_formatted_files.len(), f);
+    Ok(())
+}
+
+pub fn format(
+    config: dprint::configuration::Configuration,
+    target_files: Vec<PathBuf>,
+    check: bool,
+) -> Result<(), std::io::Error> {
+    if check {
+        check_source_files(config, target_files)?;
+    } else {
+        format_source_files(config, target_files)?;
+    }
+    Ok(())
+}
+
+fn create_cli_parser<'a, 'b>() -> clap::App<'a, 'b> {
+    App::new("dprint")
+        .about("Format source files")
+        .long_about(
+            "Auto-format JavaScript/TypeScript source code.
+  
+  dprint myfile1.ts myfile2.ts
+  
+  dprint --check myfile1.ts myfile2.ts
+  
+  dprint --config dprint.conf.json myfile1.ts myfile2.ts",
+        )
+        .arg(
+            Arg::with_name("check")
+                .long("check")
+                .help("Check if the source files are formatted.")
+                .takes_value(false),
+        )
+        .arg(
+            Arg::with_name("config")
+                .long("config")
+                .help("Path to JSON configuration file.")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("files")
+                .help("List of file paths to format")
+                .takes_value(true)
+                .multiple(true)
+                .required(true),
+        )
+}
+
+fn main() {
+    let cli_parser = create_cli_parser();
+
+    let matches = cli_parser.get_matches();
+
+    let config = if let Some(config_path) = matches.value_of("config") {
+        let config_contents = match fs::read_to_string(&config_path) {
+            Ok(contents) => contents,
+            Err(e) => {
+                eprintln!("{}", e.to_string());
+                std::process::exit(1);
+            }
+        };
+
+        let c: Configuration = match serde_json::from_str(&config_contents) {
```

Comment:
Is there a way to deserialize the object to a list of key value pairs (`HashMap<String, String>`)? Then we could just get a configuration using `resolve_config` (https://docs.rs/dprint-plugin-typescript/0.9.5/dprint_plugin_typescript/configuration/fn.resolve_config.html) and output any diagnostics.

Reply:
Seems like it's very easy with serde actually: https://github.com/serde-rs/serde/issues/1387#issuecomment-420889991